lib/datadog/appsec/processor.rb

```diff
@@ -0,0 +1,113 @@
+# typed: ignore
+
+require 'datadog/appsec/assets'
+
+module Datadog
+  module AppSec
+    # Processor integrates libddwaf into datadog/appsec
+    class Processor
+      # Interface object to check using case .. when
+      module IOLike
+        def read; end
+        def rewind; end
+
+        def self.===(other)
+          instance_methods.reduce(true) { |ok, meth| ok && other.respond_to?(meth) }
+        end
+      end
+
+      def initialize
+        unless load_libddwaf && load_ruleset && create_waf_handle
+          Datadog.logger.warn { 'appsec is disabled, cause: errors' }
+        end
+      end
+
+      def ready?
+        !@ruleset.nil? && !@handle.nil?
+      end
```

Comment:
Following the control flow, if there's errors in initialization, `@ruleset` and `@handle` never get assigned to, which means that if a customer is running with warnings on, they'll get a warning about an instance variable used before it's initialized.

Have you considered initializing them to `nil` in `#initialize`?

Reply:
Done